wger/core/views/misc.py

```diff
@@ -142,6 +143,35 @@ def dashboard(request):
     return render(request, 'index.html', template_data)
 
 
+@login_required
+def fitbitLogin(request):
+    fitbit = FitBit()
+    login_url = fitbit.ComposeAuthorizationuri()
+    return redirect(login_url)
+
+
+@login_required
+def fitbitFetch(request):
+    code = request.GET.get('code')
+    fitbit = FitBit()
+    token = fitbit.RequestAccessToken(code)
+    try:
+        data = fitbit.GetWeight(token)
+        for weight in data['weight']:
+
+            weight_entry = WeightEntry()
+            weight_entry.user = request.user
+            weight_entry.weight = weight['weight']
+            weight_entry.date = weight['date']
+            try:
+                weight_entry.save()
+            except Exception as e:
+                pass
+    except Exception as e:
+        pass
+    return HttpResponseRedirect(reverse('core:dashboard'))
```

Comment:
python pep20 errors should never pass silently 
You could print the error to the cli or log it somewhere